src/components/Article.jsx

```diff
@@ -1,28 +1,106 @@
-import { getArticle } from "../../api";
+import { getArticle, patchArticle } from "../../api";
 import React, { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import moment from "moment";
 import Comments from "./Comments";
+import { GrLike, GrDislike } from "react-icons/gr";
 
 const Article = () => {
   const { article_id } = useParams();
   const [article, setArticle] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState(null);
+  const [votes, setVotes] = useState(0);
+  const [myVote, setMyVote] = useState(null);
+  const [buttonDislikeHighlighted, setButtonDislikeHighlighted] =
+    useState(null);
+  const [buttonLikeHighlighted, setButtonLikeHighlighted] = useState(null);
 
   useEffect(() => {
     setError(null);
     getArticle(article_id)
       .then((data) => {
         setArticle(data.article);
+        setVotes(data.article.votes);
         setIsLoading(false);
       })
       .catch((error) => {
         setError("error fetching data");
         setIsLoading(false);
       });
+
+    const savedVotes = localStorage.getItem(`votes_${article_id}`);
```

Comment:
This is an interesting way to ensure likes per user are stored. You have now added things to users local storage with no way of removing it. If you wanted to make this functionality work with your api, as an extra feature you could look into how you would like up/down votes to users to stop them spamming up and down.